File: Algorithms/Island Perimeter.py
class Solution:

    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        def ajSum(i, j):
            adjcent = (i+1,j), (i-1,j), (i,j+1), (i,j-1),
            count = 0

            for x,y in adjcent:
                if x < 0 or y < 0 or x == len(grid[0]) or y == len(grid[0]) or grid[x][y] == 0:
                    count += 1
            return count

        if not grid:
            return 0

        res = 0

        for i in grid:
            for j in i:
                if j:
                    res += ajSum(i,j)

        return res

File: Algorithms/Island Perimeter.c
int islandPerimeter(int** grid, int r, int c) {
    int res = 0, i, j;

    if (!grid) {
        return 0;
    }

    for (i = 0; i < r; i++) {
        for (j = 0; j < c; j++) {
            if (grid[i][j] == 1) {
                if (i+1 == r || grid[i+1][j] == 0) {
                    res += 1;
                }

                if (i-1 < 0 || grid[i-1][j] == 0) {
                    res += 1;
                }

                if (j+1 == c || grid[i][j+1] == 0) {
                    res += 1;
                }

                if (j-1 < 0 || grid[i][j-1] == 0) {
                    res += 1;
                }
            }
        }
    }

    return res;

}
